### ISSUE TITLE
Hyperformula development installation issue on M1, M2-chip MacBooks due to incompatible Canvas package

### ISSUE THREAD

**Issue by aninde** [handsontable/hyperformula#1166]
### Description
Hyperformula depends on the `Canvas` package, [which is not currently compatible with the `arm64` architecture used by M1 and M2-chip MacBooks](https://github.com/Automattic/node-canvas/issues/2193). As a result, it is impossible to install Hyperformula using "npm i" or "npm ci" on these devices.  The problem does not affect users, only people who want to install and build Hyperformula directly on new macbooks.

**I was able to install hyperformula in project by `npm i hyperformula` on my macbook with m2 chip.**

Although `canvas` is listed as a peerDependency for `jsdom`, it is not used in the Hyperformula codebase.

To solve the installation problem on macOS with M1 chips, I removed `canvas` from the `package.json` file. This allowed me to successfully build the package, and all of the following scripts passed without issue:

```
//check system architecture
arch

npm i
npm run bundle-all
npm run verify-bundles
npm run test
npm run test:unit
npm run test:coverage
npm run test:browser
```

After careful research of the use of `canvas`, I propose to remove it from the list of dependencies.

### Your environment

macOS Apple M2

* HyperFormula version: -
* Browser Name and version: -
* Operating System: 13.0 (22A380)

### Links

Logs:
```
npm ERR! code 1
npm ERR! path /Users/nataliademytruk/Projects/hyperformula/node_modules/canvas
npm ERR! command failed
npm ERR! command sh -c -- node-pre-gyp install --fallback-to-build --update-binary
npm ERR! Failed to execute '/Users/nataliademytruk/.nvm/versions/node/v16.19.0/bin/node /Users/nataliademytruk/.nvm/versions/node/v16.19.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --update-binary --module=/Users/nataliademytruk/Projects/hyperformula/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/Users/nataliademytruk/Projects/hyperformula/node_modules/canvas/build/Release --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v93' (1)
npm ERR! node-pre-gyp info it worked if it ends with ok
npm ERR! node-pre-gyp info using node-pre-gyp@1.0.10
npm ERR! node-pre-gyp info using node@16.19.0 | darwin | arm64
npm ERR! node-pre-gyp http GET https://github.com/Automattic/node-canvas/releases/download/v2.10.2/canvas-v2.10.2-node-v93-darwin-unknown-arm64.tar.gz
npm ERR! node-pre-gyp ERR! install response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.10.2/canvas-v2.10.2-node-v93-darwin-unknown-arm64.tar.gz
npm ERR! node-pre-gyp WARN Pre-built binaries not installable for canvas@2.10.2 and node@16.19.0 (node-v93 ABI, unknown) (falling back to source compile with node-gyp)
npm ERR! node-pre-gyp WARN Hit error response status 404 Not Found on https://github.com/Automattic/node-canvas/releases/download/v2.10.2/canvas-v2.10.2-node-v93-darwin-unknown-arm64.tar.gz
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.1.0
npm ERR! gyp info using node@16.19.0 | darwin | arm64
npm ERR! gyp info ok
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.1.0
npm ERR! gyp info using node@16.19.0 | darwin | arm64
npm ERR! gyp info find Python using Python version 3.9.6 found at "/Library/Developer/CommandLineTools/usr/bin/python3"
npm ERR! gyp info spawn /Library/Developer/CommandLineTools/usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/Users/nataliademytruk/.nvm/versions/node/v16.19.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/nataliademytruk/Projects/hyperformula/node_modules/canvas/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/nataliademytruk/.nvm/versions/node/v16.19.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/nataliademytruk/Library/Caches/node-gyp/16.19.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/Users/nataliademytruk/Library/Caches/node-gyp/16.19.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/Users/nataliademytruk/.nvm/versions/node/v16.19.0/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/Users/nataliademytruk/Library/Caches/node-gyp/16.19.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/Users/nataliademytruk/Projects/hyperformula/node_modules/canvas',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! /bin/sh: pkg-config: command not found
npm ERR! gyp: Call to 'pkg-config pixman-1 --libs' returned exit status 127 while in binding.gyp. while trying to load binding.gyp
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack     at ChildProcess.onCpExit (/Users/nataliademytruk/.nvm/versions/node/v16.19.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:284:16)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)
npm ERR! gyp ERR! System Darwin 22.1.0
npm ERR! gyp ERR! command "/Users/nataliademytruk/.nvm/versions/node/v16.19.0/bin/node" "/Users/nataliademytruk/.nvm/versions/node/v16.19.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--update-binary" "--module=/Users/nataliademytruk/Projects/hyperformula/node_modules/canvas/build/Release/canvas.node" "--module_name=canvas" "--module_path=/Users/nataliademytruk/Projects/hyperformula/node_modules/canvas/build/Release" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v93"
npm ERR! gyp ERR! cwd /Users/nataliademytruk/Projects/hyperformula/node_modules/canvas
npm ERR! gyp ERR! node -v v16.19.0
npm ERR! gyp ERR! node-gyp -v v9.1.0
npm ERR! gyp ERR! not ok
npm ERR! node-pre-gyp ERR! build error
npm ERR! node-pre-gyp ERR! stack Error: Failed to execute '/Users/nataliademytruk/.nvm/versions/node/v16.19.0/bin/node /Users/nataliademytruk/.nvm/versions/node/v16.19.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --update-binary --module=/Users/nataliademytruk/Projects/hyperformula/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/Users/nataliademytruk/Projects/hyperformula/node_modules/canvas/build/Release --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v93' (1)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/nataliademytruk/Projects/hyperformula/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1100:16)
npm ERR! node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
npm ERR! node-pre-gyp ERR! System Darwin 22.1.0
npm ERR! node-pre-gyp ERR! command "/Users/nataliademytruk/.nvm/versions/node/v16.19.0/bin/node" "/Users/nataliademytruk/Projects/hyperformula/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--update-binary"
npm ERR! node-pre-gyp ERR! cwd /Users/nataliademytruk/Projects/hyperformula/node_modules/canvas
npm ERR! node-pre-gyp ERR! node -v v16.19.0
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.10
npm ERR! node-pre-gyp ERR! not ok
npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/nataliademytruk/.npm/_logs/2023-02-28T13_32_12_168Z-debug-0.log
```

**Comment by thilgen** [comment]
** **Another M1 Mac User Here** ** 

I have successfully (and very happily) been using Hyperformula in my project by installing the package in my project with `npm install hyperformula` 👍 

But when I decided to try and build the `handsontable/hyperformula` project locally to see if I could add a new feature (following the information on the [Building Hyperformula](https://hyperformula.handsontable.com/guide/building.html#build-the-project) page
```
$ git clone git@github.com:handsontable/hyperformula.git
$ cd hyperformula
$ npm install 
```
The install failed to complete due to the `canvas` issue mentioned above (same output / same root cause / M1 Mac plus older versions required packages)

My solution was the following

1. Install the latest version of `node-gyp` globally
2. Follow the [instructions for compiling `node-canvas`](https://github.com/Automattic/node-canvas) on OS X using Homebrew
4. Successfully run install for the `handsontable/hyperformula` project
```
$ sudo npm install -g node-gyp
$ brew install pkg-config cairo libpng jpeg giflib pango librsvg pixman
$ npm install
```

I don't believe that this is the best long-term solution...

**but until the `handsontable/hyperformula` package list can be updated to work well on M1 machines - would it be possible to add a note to [Building Hyperformula](https://hyperformula.handsontable.com/guide/building.html#build-the-project) that explains what is going on and to include a workaround?**

Would have definitely saved me 3-4 hours of searching around for a solution.

**Comment by AMBudnik** [comment]
@thilgen we are aiming to fix this issue in the upcoming release. I will surely notify you when that happens.

**Comment by aninde** [comment]
I made sure that on my device it can be built locally with develop version 76561e2196177fe1b4ed42cb049eb1bdd5169911.
It works after `npm i`and `npm ci`.  I was working on a clean environment, newly downloaded repository on Macbook Air Apple M2 2022 macOS Ventura 13.3.1 (22E261).

**Comment by AMBudnik** [comment]
Issue closed as solved and releases in Hyperformula 2.4.0

**Comment by sequba** [comment]
@thilgen ⬆️ :)
